src/ui/component/dropDownMenu/dropDownMenu.tsx

```diff
@@ -0,0 +1,159 @@
+import { useCallback, useMemo, useRef, useState } from 'react'
+import type { Dispatch, SetStateAction, FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useOnClickOutside } from '@/ui/hook/useOnClickOutside'
+import { useAppStore } from '@/ui/store/appStore'
+import { Icon } from '@/ui/component/icon/icon'
+import classnames from 'classnames'
+import './dropDownMenu.scss'
+
+type ID = string
+
+export type Option = {
+  id: ID
+  title: string
+  value: ID[]
```

Comment:
associatedIDs doesn't capture the fact that the options id is also present in it's value. 

The property name should convey two facts:
- That it's the value returned when selecting an option. 
- That the return value reflects the option ids ( [...associatedIDs, optionID] )

IMO value makes it clear it's the returned value, and the type tells us it's an array of ids. I could rename the ID type to make the value property clearer.

Reply:
I agree with your point that the `value` field conveys the information that it's the value returned when selecting an option.
My original feedback was more towards enhancing readability. When encountering `foo.value` or `foo.associatedIDs` in the code, the latter provides a more immediate understanding of what it contains: a list of associated option IDs. Perhaps we can consider a compromise like `optionIDsIncludingSelf` or `selfAndAssociatedIDs` to maintain clarity while also expressing that the option's own ID is included. I believe this would enhance the overall readability of our code.